From e2c32579130fe27818f37063f1f69a0f57d0265e Mon Sep 17 00:00:00 2001
From: Peter Marschall <peter@adpm.de>
Date: Thu, 2 Jun 2011 16:02:17 +0200
Subject: [PATCH 15/26] opensc-explorer: consolidate string parsing in
 do_update_record()

Update do_update_record() to use parse_string_or_hexdata() instead of the old
hex2binary().
This change allows to use double-quoted strings in the "update_record" command.

Signed-off-by: Peter Marschall <peter@adpm.de>
---
 src/tools/opensc-explorer.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/tools/opensc-explorer.c b/src/tools/opensc-explorer.c
index 021dc32..2735924 100644
--- a/src/tools/opensc-explorer.c
+++ b/src/tools/opensc-explorer.c
@@ -159,7 +159,7 @@ static struct command	cmds[] = {
 		"random",	"<count>",
 		"obtain <count> random bytes from card"	},
 	{ do_update_record,
-		"update_record", "<file id> <rec no> <rec offs> <hex value>",
+		"update_record", "<file id> <rec no> <rec offs> <data>",
 		"update record"				},
 	{ do_update_binary,
 		"update_binary", "<file id> <offs> <data>",
@@ -1176,6 +1176,7 @@ err:
 static int do_update_record(int argc, char **argv)
 {
 	u8 buf[240];
+	size_t buflen;
 	int r, i, err = 1;
 	int rec, offs;
 	sc_path_t path;
@@ -1212,9 +1213,10 @@ static int do_update_record(int argc, char **argv)
 		goto err;;
 	}
 
-	i = hex2binary(buf + offs, sizeof(buf) - offs, in_str);
+	buflen = sizeof(buf) - offs;
+	i = parse_string_or_hexdata(in_str, buf + offs, &buflen);
 	if (!i) {
-		printf("unable to parse hex value\n");
+		printf("unable to parse data\n");
 		goto err;
 	}
 
-- 
1.7.5.3

